localstack/aws/handlers/service_plugin.py

```diff
@@ -40,7 +41,7 @@ def require_service(self, _: HandlerChain, context: RequestContext, response: Re
             return
 
         service_name: str = context.service.service_name
-        if not self.service_manager.exists(service_name):
+        if not self.service_manager.exists(service_name) or not is_api_enabled(service_name):
             raise NotImplementedError
```

Comment:
This could be problematic in combination with the handling of the `NotImplementedError` in `skeleton.py`:
https://github.com/localstack/localstack/blob/d97a2e8b4183d7b978b61bd8583f811e5ccd392a/localstack/aws/skeleton.py#L195-L222

It would be great if the error message could be different if a service is disabled (in comparison to the error message when the service operation is just not implemented / not found at all).
We could maybe handle this by creating a specific subclass of `NotImplementedError` (maybe something like `ServiceNotEnabledError`?), and in the skeleton we could then differentiate and use a different message (other than the `MESSAGE_TEMPLATE` in `coverage_docs.py`.
Another, possibly simpler, solution would be to raise a `NotimplementedError` with the specific error message right here (which will then be used, since the message template is just a fallback).

Reply:
Right, good catch. Thomas said the services should raise `NotImplemented` if not enabled. But a different message would be great, definitely, and will help users understand what's wrong.